src/pages/profile-page/ProfilePage.jsx

```diff
@@ -0,0 +1,127 @@
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Post } from "../../components";
+import { useParams } from "react-router-dom";
+import { followUser,unFollowUser } from "../../features/user/userSlice";
+
+const ProfilePage = () => {
+  const dispatch = useDispatch();
+  const { user, token } = useSelector((state) => state.auth);
+  const { posts } = useSelector((state) => state.post);
+  const { users,status } = useSelector((state) => state.user);
+  const [numberOfPosts, setNumberOfPosts] = useState(0);
+  const [profileUser, setProfileUser] = useState({});
+  let { profileId } = useParams();
+  const [isFollowed, setIsFollowed] = useState(false);
+
+  useEffect(() => {
+    const profileUser = users.filter((user) => user.username === profileId);
+    setProfileUser(profileUser[0]);
+  }, [users, profileId]);
+
+  useEffect(() => {
+    const followed = users
+      .find((userr) => userr.username === user.username)
```

Comment:
is this a typo ? `userr`

Reply:
No, tried to differentiate it with user defined earlier.